symengine/tests/cwrapper/test_cwrapper.c

```diff
@@ -458,6 +459,213 @@ void test_ascii_art() {
+
+    basic_atan(ans, one);
+    s = basic_str(ans);
+    SYMENGINE_C_ASSERT(strcmp(s, "(1/4)*pi") == 0);
+    SYMENGINE_C_ASSERT(basic_eq(ans, pi_div_four));
+    basic_str_free(s);
+
+    basic_acot(ans, one);
+    s = basic_str(ans);
+    SYMENGINE_C_ASSERT(strcmp(s, "(1/4)*pi") == 0);
+    SYMENGINE_C_ASSERT(basic_eq(ans, pi_div_four));
+    basic_str_free(s);
+
+    basic_sinh(ans, zero);
+    s = basic_str(ans);
+    SYMENGINE_C_ASSERT(strcmp(s, "0") == 0);
+    SYMENGINE_C_ASSERT(basic_eq(ans, zero));
+    basic_str_free(s);
+
+    basic_cosh(ans, zero);
+    s = basic_str(ans);
+    SYMENGINE_C_ASSERT(strcmp(s, "1") == 0);
+    SYMENGINE_C_ASSERT(basic_eq(ans, one));
+    basic_str_free(s);
+
+    basic_tanh(ans, zero);
+    s = basic_str(ans);
+    SYMENGINE_C_ASSERT(strcmp(s, "0") == 0);
+    SYMENGINE_C_ASSERT(basic_eq(ans, zero));
+    basic_str_free(s);
+
+    basic_csch(ans, one);
+    s = basic_str(ans);
+    SYMENGINE_C_ASSERT(strcmp(s, "csch(1)") == 0);
+    basic_str_free(s);
+
+    basic_sech(ans, zero);
+    s = basic_str(ans);
+    SYMENGINE_C_ASSERT(strcmp(s, "1") == 0);
+    SYMENGINE_C_ASSERT(basic_eq(ans, one));
+    basic_str_free(s);
+
+    basic_coth(ans, one);
+    s = basic_str(ans);
+    SYMENGINE_C_ASSERT(strcmp(s, "coth(1)") == 0);
+    basic_str_free(s);
+
+    basic_asinh(ans, zero);
+    s = basic_str(ans);
+    SYMENGINE_C_ASSERT(strcmp(s, "0") == 0);
+    SYMENGINE_C_ASSERT(basic_eq(ans, zero));
+    basic_str_free(s);
+
+    basic_acosh(ans, one);
+    s = basic_str(ans);
+    SYMENGINE_C_ASSERT(strcmp(s, "0") == 0);
+    SYMENGINE_C_ASSERT(basic_eq(ans, zero));
+    basic_str_free(s);
+
+    basic_tanh(ans, zero);
+    s = basic_str(ans);
+    SYMENGINE_C_ASSERT(strcmp(s, "0") == 0);
+    SYMENGINE_C_ASSERT(basic_eq(ans, zero));
+    basic_str_free(s);
+
+    basic_acsch(ans, one);
+    s = basic_str(ans);
+    SYMENGINE_C_ASSERT(strcmp(s, "log(1 + 2**(1/2))") == 0);
+    basic_str_free(s);
+
+    basic_asech(ans, one);
+    s = basic_str(ans);
+    SYMENGINE_C_ASSERT(strcmp(s, "0") == 0);
+    SYMENGINE_C_ASSERT(basic_eq(ans, zero));
+    basic_str_free(s);
+
+    basic_acoth(ans, one);
+    s = basic_str(ans);
+    SYMENGINE_C_ASSERT(strcmp(s, "acoth(1)") == 0);
+    basic_str_free(s);
+
+    basic_lambertw(ans, e_minus_one);
+    s = basic_str(ans);
+    SYMENGINE_C_ASSERT(strcmp(s, "lambertw(-1/E)") == 0);
+    //SYMENGINE_C_ASSERT(basic_eq(ans, minus_one));
+    basic_str_free(s);
+
+    basic_zeta(ans, zero);
+    s = basic_str(ans);
+    SYMENGINE_C_ASSERT(strcmp(s, "-1/2") == 0);
+    SYMENGINE_C_ASSERT(basic_eq(ans, minus_half));
+    basic_str_free(s);
+
+    basic_dirichlet_eta(ans, one);
+    s = basic_str(ans);
+    SYMENGINE_C_ASSERT(strcmp(s, "log(2)") == 0);
+    basic_str_free(s);
+
+    basic_gamma(ans, one);
+    s = basic_str(ans);
+    SYMENGINE_C_ASSERT(strcmp(s, "1") == 0);
+    SYMENGINE_C_ASSERT(basic_eq(ans, one));
+    basic_str_free(s);
+
+    basic_gamma(ans, two);
```

Comment:
We won't need two tests for `basic_gamma`. Exhaustive tests are already written in the C++ source. We only need to check if the function works for any one value.